airflow/utils/sqlalchemy.py

```diff
@@ -166,9 +166,10 @@ def bind_processor(self, dialect):
         super_process = super().bind_processor(dialect)
 
         def process(value):
-            if isinstance(value, dict) and 'pod_override' in value:
-                value['pod_override'] = BaseSerialization()._serialize(value['pod_override'])
-            return super_process(value)
+            val_copy = copy.copy(value)
+            if isinstance(val_copy, dict) and 'pod_override' in val_copy:
+                val_copy['pod_override'] = BaseSerialization._serialize(val_copy['pod_override'])
```

Comment:
Would it be more reliable to check for `__type` and `__var` instead? (We could have that logic in `serialized_objects` instead as a `is_serialized` function)

Reply:
I’m thinking line 170 should be

```python
if not is_serialized(val_copy):
```

with the function implemented as

```python
def is_serialized(v):
    if not isinstance(v, dict):
        return False
    return Encoding.TYPE in v and Encoding.VAR in v
```

This can be used to unify the logic on line 187/188.

Reply:
so the thing is, re 170, i see what you are suggesting. but the reason we are checking specifically for pod_override here, is because we're not trying to serialize any and all executor configs.  in general, they are simply pickled.  but because of idiosyncrasies of the design of the k8s objects, it's safer to essentially convert them to json before pickling , which in effect is what the serializer does.  basically, we just want to do this for k8s executor configs.  if we wanted to do for all configs, your suggestion makes complete sense.